### PR TITLE
DAC6-3001: Allow requests from Agent affinity group

### PR DESCRIPTION
**Pull request by koladeadewuyi**
This PR allows requests having an agent affinity group to access the service as per the requirement below from [DAC6-3001](https://jira.tools.tax.service.gov.uk/browse/DAC6-3001):

```
There are 2 change contact details journeys and the version shown is dependent on the user's affinity group:

- organisation, for users with an affinity group of organisation or agent
- individual, for users with an affinity group of individual
```

### CODE REVIEW

File: app/uk/gov/hmrc/crsfatcaregistration/auth/AuthAction.scala [hunk 1]
```diff
@@ -63,8 +63,7 @@ class AuthActionImpl @Inject() (
     credentialRole: Option[CredentialRole]
   ): Boolean =
     affinityGroup match {
-      case Some(Agent) => false
-      case Some(Organisation) =>
+      case Some(Organisation) | Some(Agent) =>
         credentialRole.fold(false)(
```
Comment (chriswaugh) [inline app/uk/gov/hmrc/crsfatcaregistration/auth/AuthAction.scala:67]: What is this credential role check for? Is it correct / applicable? For agents and organisations?

Comment (koladeadewuyi) [reply app/uk/gov/hmrc/crsfatcaregistration/auth/AuthAction.scala:67]: After discussion with @jagankumar-hmrc, these checks ought to be on the functionality level and will be ensured on the frontend not on the backend.
I'll push a commit to remove the checks from the backend and will implement them on the frontend instead.

Comment (chriswaugh) [reply app/uk/gov/hmrc/crsfatcaregistration/auth/AuthAction.scala]: Sorry Kolade, I only meant to better understand what this credential check was for.

I've done some digging and if I'm right it's only Admins who are able to register for the service. Regular users can submit reports but not register. Is that right?

If so I think we need to check for this in the frontend AND in the backend. A check in the backend to make sure we're secure and not doing the wrong thing, and a check in the frontend so we can fail fast and present an appropriate error to the user without needing to call the backend.

Comment (koladeadewuyi) [reply app/uk/gov/hmrc/crsfatcaregistration/auth/AuthAction.scala]: @chriswaugh that's what I resorted to as well. The frontend does appropriate checks. The backend also does its own checks. The registration APIs are only accessible to admins while the subscription APIs are accessible to all.

Comment (koladeadewuyi) [reply app/uk/gov/hmrc/crsfatcaregistration/auth/AuthAction.scala]: @chriswaugh the authentication has now been updated.

**Review by chriswaugh — APPROVED** [review]

**Pull request merged**